### ISSUE TITLE
Time to Charge: Fully Charged is incorrect

### ISSUE THREAD

**Issue by DePingus** [dubrowgn/wattz#33]
Hi. Nice app!

Where it says "Time to Charge" it says that the phone is "fully charged". My battery is currently at 76% according to the battery page of the settings. So that's not being calculated correctly.

I'm on a stock Google Pixel 7. The app does correctly detect when the phone is plugged into an AC charger, a computer's USB port, or not plugged into anything at all.

**Comment by dubrowgn** [comment]
Hi, thanks for reporting the issue. Would it be possible to get a screenshot of the main wattz view when it's doing this? If there is anything in that field, it should just be reporting what the OS is reporting. A screenshot would help show me a full picture of what the battery management system is reporting at the time.

**Comment by EvanSunde** [comment]
Hello dubrowgn, 

I am facing the same problem

I have attached the screenshot ![Screenshot_2024-04-01-19-19-06-64_89bc7522f6dc9ac5cd7abb34f6bc1ada.jpg](https://github.com/dubrowgn/wattz/assets/77535604/4b97c2b9-0457-42b3-b1c7-ff61b85cc999)

![Screenshot_2024-04-01-19-19-01-01_89bc7522f6dc9ac5cd7abb34f6bc1ada.jpg](https://github.com/dubrowgn/wattz/assets/77535604/f13412a3-301e-4615-9a7c-8f5612991d7d)

**Comment by dubrowgn** [comment]
Thanks for the screen shots. If there were any issues collecting the "time to full charge" metric from the OS, the field would contain a dash instead of "fully charged". This means the OS is successfully telling the app there is zero seconds left until full charge.

@DePingus, @kevnil0420: do either of you also need to use the "Charging Workaround"? This hasn't been necessary on older Samsung and Pixel devices, so it may indicate something has changed in the way this data is reported. "Is charging", for example, is actually reported in more than one way, so we potentially could try one of the other ones.

For charge time remaining, unfortunately I'm not aware of another API to query for this information.

Have any of you changed the batteries in your phone by chance?

**Comment by kevnil0420** [comment]
Yea I need to use it

**Comment by EvanSunde** [comment]
@dubrowgn no, it's the stock battery. And after turning off the charging workaround, the watt is correct with the correct negative and positive symbol, but it still does not show time to fully charge

**Comment by dubrowgn** [comment]
Interestingly, I saw something like this on my device (Pixel 5) the other day on my lock screen. I unlocked the phone and it snapped back to normal.

**Comment by eapo** [comment]
Same here on stock Pixel 7 Pro

![Screenshot_20240916-004307](https://github.com/user-attachments/assets/7abbce98-3a37-43bf-b16c-96b6b193b138)

**Comment by extioStorm** [comment]
Couldn't wattz calculate itself in the situation?  If you have graphed the time it took to charge previously, and know when ever the voltage will step down, then you should be able to calculate it yourself

**Comment by bojogon7** [comment]
Same pixel 8 pro
